File: week_1/day_1_arrays/move_zero_follow_up_questions/Stable_vs_Unstable_Approach.js
function moveZeroesStable(nums) {
    let j = 0;
    for (let i = 0; i < nums.length; i++) {
        if (nums[i] !== 0) {
            nums[j++] = nums[i];
        }
    }
    while (j < nums.length) {
        nums[j++] = 0;
    }
}

let nums = [0, 1, 0, 3, 12];
moveZeroesStable(nums);
console.log(nums);

File: week_1/day_1_arrays/move_zero_follow_up_questions/Stable_vs_Unstable_Approach.py
def moveZeroesStable(nums):
    j = 0
    for i in range(len(nums)):
        if nums[i] != 0:
            nums[j] = nums[i]
            j += 1
    while j < len(nums):
        nums[j] = 0
        j += 1

nums = [0, 1, 0, 3, 12]
moveZeroesStable(nums)
print(nums)

